File: java/GolfBall/src/Club.java
/**
 * @author 
 * {@literal} Description : Class de definition d'un Club
 */
public class Club {
	private String Type;
	private double Poids;
	private double Loft;
	private double Ecoeff;
	private double CoeffBackSpin;
	private double CoeffSpinLift;
	private double Cl1;					// A new aerodynamic model of a golf ball in flight (1994); Alexander J Smits
	private double alphaClubPath;		// angle du chemin de club a l'impact
	private double gamaFacePath;		// angle de la tete de club a l'impact
	private double shaftLeanImp;			// angle du shaft a l'impact en degre
	private double clubV0kmh;			// vitesse en km/h dde la tete de club
	private double miss;				// centre du club loupe en cm
	private double temperature;	 	// Temperature de l'aire en °C

	// Constructor 
	public Club(String pType, double pPoids, double pTemperature, double pLoft, double pEcoeff, double pCoeffBackSpin, double pCoeffSpinLift, double pCl1, double pClubV0, double pAlphaClubPath, double pGamaFacePath, double pShaftLeanImp){
		/**
		 * @author 
		 * @param String pType, double pPoids, double pLoft, double pEcoeff, double pCoeffBackSpin, double pCoeffSpinLift, double pCm)
		 * 
		 */
	    System.out.println("Création d\'un Club : " + pType);
		Type 			= pType;
		Poids 			= pPoids;
		Loft 			= pLoft; // en degre
		Ecoeff 			= pEcoeff;
		CoeffBackSpin	= pCoeffBackSpin;
		CoeffSpinLift 	= pCoeffSpinLift;
		Cl1 			= pCl1;
		alphaClubPath 	= pAlphaClubPath*Math.PI/180; // en radian
		gamaFacePath 	= pGamaFacePath*Math.PI/180;  // en radian
		shaftLeanImp 	= pShaftLeanImp;
		clubV0kmh		= pClubV0;
		temperature 	= pTemperature;
		
	} // fin constructor public GolfClub()
	
	public String getType() {
		return Type;
	}	
	public double getTemperature() {
		return temperature;
	}
	public void setTemperature(double temp) {
		temperature = temp;
	}
	public double getPoids() {
		return Poids;
	}
	public void setPoids(double poids) {
		Poids = poids;
	}
	public double getEcoeff() {
		return Ecoeff;
	}
	public void setECoeff(double pEcoeff) {
		Ecoeff = pEcoeff;
	}
	public double getCoeffBackSpin() {
		return CoeffBackSpin;
	}
	public void setCoeffBackSpin(double pCoeffBackSpin) {
		CoeffBackSpin = pCoeffBackSpin;
	}
	public double getCoeffSpinLift() {
		return CoeffSpinLift;
	}
	public void setCoeffSpinLift(double pCoeffSpinLift) {
		CoeffSpinLift = pCoeffSpinLift;
	}
	public double getCl1() {
		return Cl1;
	}
	public void setCl1(double pCl1) {
		Cl1 = pCl1;
	}
	public double getLoft() {
		return Loft;
	}
	public void setLoftDegre(double pLoft) {
		Loft = pLoft;
	}
	public double getDynamiqueLoftRadian() {
		return (getLoft() + getShaftLeanImp())*Math.PI/180;
	}
	public double getDynamiqueLoftDegre() {
		return getDynamiqueLoftRadian()*180/Math.PI;
	}
	public void setShafLeanImpDegre(double pShafLeanImp) {
		shaftLeanImp = pShafLeanImp;
	}
	public double getShaftLeanImp() {
		return shaftLeanImp;
	}
	public void setShaftLeanImpDegre(double pShaftLean) {
		shaftLeanImp = pShaftLean;
	}
	public void setClubV0Kmh(double vitesse) {
		clubV0kmh	= vitesse;
	}
	public double getClubV0kmh() {
		return clubV0kmh;
	}	
	public double getClubV0ms() {
		return getClubV0kmh() * 10 / 36;
	}
	public double getMiss() {
		return miss;
	}
	public double getGamaFacePathRadian() {
		return gamaFacePath;
	}
	public void setGamaFacePathDegre(double pGamaFacePath) {
		gamaFacePath = pGamaFacePath*Math.PI/180;
	}
	public double getAlphaClubPathRadian() {
		return alphaClubPath;
	}
	public void setAlphaClubPathDegre(double pAlphaClubPath) {
		alphaClubPath = pAlphaClubPath*Math.PI/180;
	}
} // fin public class GolfClub

File: Cpp/src/Club.cpp
/*
 * Club.cpp
 *
 *  Created on: 19 sept. 2016
 *      Author: matthieu
 */
#include <string>
#include "Club.h"

using namespace std;

Club::Club():
				Type("Dr"), Poids(0.300), temperature(20.0), Loft(11.0), Ecoeff(0.738), CoeffBackSpin(200.00), CoeffSpinLift(1500.0),
				Cl1(0.64), clubV0kmh(170.6), alphaClubPath(0.0), gamaFacePath(0.0), shaftLeanImp(6.0){};

Club::Club(const std::string pType, double pPoids, double pTemperature, double pLoft, double pEcoeff, double pCoeffBackSpin,
		double pCoeffSpinLift, double pCl1, double pClubV0, double pAlphaClubPath, double pGamaFacePath,
		double pShaftLeanImp):
		Type(pType), Poids(pPoids), temperature(pTemperature), Loft(pLoft), Ecoeff(pEcoeff), CoeffBackSpin(pCoeffBackSpin), CoeffSpinLift(pCoeffSpinLift),
		Cl1(pCl1), clubV0kmh(pClubV0), alphaClubPath(pAlphaClubPath*M_PI/180.0), gamaFacePath(pGamaFacePath*M_PI/180.0), shaftLeanImp(pShaftLeanImp),miss(0) {}
Club::~Club() {
}

std::string Club::getType() {
	return Type;
}
double Club::getTemperature() {
	return temperature;
}
void Club::setTemperature(double temp) {
	temperature = temp;
}
double Club::getPoids() {
	return Poids;
}
void Club::setPoids(double poids) {
	Poids = poids;
}
double Club::getEcoeff() {
	return Ecoeff;
}
void Club::setECoeff(double pEcoeff) {
	Ecoeff = pEcoeff;
}
double Club::getCoeffBackSpin() {
	return CoeffBackSpin;
}
void Club::setCoeffBackSpin(double pCoeffBackSpin) {
	CoeffBackSpin = pCoeffBackSpin;
}
double Club::getCoeffSpinLift() {
	return CoeffSpinLift;
}
void Club::setCoeffSpinLift(double pCoeffSpinLift) {
	CoeffSpinLift = pCoeffSpinLift;
}
double Club::getCl1() {
	return Cl1;
}
void Club::setCl1(double pCl1) {
	Cl1 = pCl1;
}
double Club::getLoft() {
	return Loft;
}
void Club::setLoftDegre(double pLoft) {
	Loft = pLoft;
}
double Club::getDynamiqueLoftRadian() {
	return (getLoft() + getShaftLeanImp())*M_PI/180.0;
}
double Club::getDynamiqueLoftDegre() {
	return getDynamiqueLoftRadian()*180.0/M_PI;
}
double Club::getShaftLeanImp() {
	return shaftLeanImp;
}
void Club::setShaftLeanImpDegre(double pShaftLean) {
	shaftLeanImp = pShaftLean;
}
void Club::setClubV0Kmh(double vitesse) {
	clubV0kmh	= vitesse;
}
double Club::getClubV0kmh() {
	return clubV0kmh;
}
double Club::getClubV0ms() {
	return getClubV0kmh() * 10.0 / 36.0;
}
double Club::getMiss() {
	return miss;
}
double Club::getGamaFacePathRadian() {
	return gamaFacePath;
}
void Club::setGamaFacePathDegre(double pGamaFacePath) {
	gamaFacePath = pGamaFacePath*M_PI/180.0;
}
double Club::getAlphaClubPath() {
	return alphaClubPath;
}
double Club::getAlphaClubPathRadian() {
	return alphaClubPath;
}
void Club::setAlphaClubPathDegre(double pAlphaClubPath) {
	alphaClubPath = pAlphaClubPath*M_PI/180.0;
}


